hudi-hive/src/main/java/org/apache/hudi/hive/HoodieHiveClient.java

```diff
@@ -632,24 +633,33 @@ public void close() {
     if (!lastCommitTimeSynced.isPresent()) {
       LOG.info("Last commit time synced is not known, listing all partitions in " + syncConfig.basePath + ",FS :" + fs);
       try {
-        return FSUtils.getAllPartitionPaths(fs, syncConfig.basePath, syncConfig.assumeDatePartitioning);
+
+        List<String> fsPartitions = FSUtils.getAllPartitionPaths(fs, syncConfig.basePath, syncConfig.assumeDatePartitioning);
+        List<String> tlPartitions = findPartitionsAfter(HOODIE_FIRST_COMMIT_TIME);
```

Comment:
This code is attempting to union the results from both `FSUtils.getAllPartitionPaths` and the timeline search.. But we only keep the last N commit files in the active timeline (rest gets archived).. So, there could be some partitions written in these archived commits that the `findPartitionsAfter` method will not inspect.. What I am saying is, to properly handle all cases, we need FSUtils.getAllPartitionPaths to work anyway.
 
So, this is not guaranteed to fix the issue anyway. if the `assumeDatePartitioning` is set to true (false by default), we can just assume the user knows what he/she is doing and ignore this?

Reply:
Hi @vinothchandar, I know what you mean. The problem the user is experiencing is can not get any data at the first time, this solution is union results from both `FSUtils.getAllPartitionPaths` and the first commit.